vector/src/main/codegen/templates/PromotableWriter.java

```diff
@@ -540,18 +543,25 @@ public void writeLargeVarChar(String value) {
     getWriter(MinorType.LARGEVARCHAR).writeLargeVarChar(value);
   }
 
+  protected ArrowType lastExtensionType;
+
   @Override
   public void writeExtension(Object value) {
```

Comment:
(design) should we deprecate this method? (since we now have `writeExtension(ExtensionHolder)`